### PR TITLE
Fix auth redirection, some refactoring

### PR DESCRIPTION
**Pull request by feedmeapples**
- fixes issue of when Web would prioritize `defaultToNamespace` config redirection over signin page (if not signed in)
- moved `auth.enabled` config value from `api/me` to `api/web-settings`

### CODE REVIEW

File: client/App.vue [hunk 1]
```diff
@@ -121,35 +114,44 @@ export default {
...
-
-      if (defaultToNamespace !== this.$route.params.namespace) {
-        this.onNotification({
-          message: `No access to namespace ${this.$route.params.namespace}. Redirecting to ${defaultToNamespace}`,
-          type: NOTIFICATION_TYPE_ERROR,
-        });
-        this.$router.push(`/namespaces/${defaultToNamespace}/workflows`);
-      }
+    async getWebSettings() {
+      this.webSettings = await this.$http(`/api/web-settings`);
```
Comment (rylandg) [inline client/App.vue:118]: Why is this a template string?

File: client/App.vue [hunk 2]
```diff
@@ -121,35 +114,44 @@ export default {
...
+    async getCurrentUser() {
       const me = await this.$http('/api/me');
-      if (me.isAuthEnabled) {
-        if (!me.user) {
+      this.currentUser = me?.user;
+    },
+    redirectIfApplicable() {
+      const { auth, routing } = this.webSettings;
+
+      if (auth?.enabled) {
```
Comment (rylandg) [inline client/App.vue:127]: nit: this could be simplified

```js
if (auth?.enabled && !this.currentUser) 
```

File: client/App.vue [hunk 3]
```diff
@@ -121,35 +114,44 @@ export default {
...
+        if (!this.currentUser) {
           this.$router.push('/signin');
-        } else {
-          this.currentUser = me.user;
+          return;
         }
       }
+      if (routing?.defaultToNamespace) {
```
Comment (rylandg) [inline client/App.vue:133]: This could also be simplified.

File: client/App.vue [hunk 4]
```diff
@@ -121,35 +114,44 @@ export default {
...
+          });
+          this.$router.push(`/namespaces/${defaultToNamespace}/workflows`);
+          return;
+        }
+      }
+    },
+    async announceNewVersionIfExists() {
+      const { show, message, link, version } = await getNewVersionAnnouncement(
+        this.$http,
+        this.onNotification
```
Comment (rylandg) [inline client/App.vue:148]: nit: trailing comma

**Review by rylandg — APPROVED** [review]
A few nits

File: client/App.vue (hunk 4 above)
Comment (feedmeapples) [reply client/App.vue:148]: this is not currently in the proj's prettier config

File: client/App.vue (hunk 1 above)
Comment (feedmeapples) [reply client/App.vue]: addressed

File: client/App.vue (hunk 3 above)
Comment (feedmeapples) [reply client/App.vue]: addressed

File: client/App.vue (hunk 2 above)
Comment (feedmeapples) [reply client/App.vue]: addressed

**Pull request merged**